internal/pkg/agent/application/coordinator/coordinator_unit_test.go

```diff
@@ -520,6 +522,137 @@ inputs:
+	// another runtime manager update with an empty component model.
+
+	// Set a one-second timeout -- nothing here should block, but if it
+	// does let's report a failure instead of timing out the test runner.
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+	defer cancel()
+	logger := logp.NewLogger("testing")
+
+	configChan := make(chan ConfigChange, 1)
+
+	// Create a mocked runtime manager that will report the update call
+	runtimeManager := &fakeRuntimeManager{
+		updateCallback: func(comp []component.Component) error {
+			return nil
+		},
+	}
+
+	monitoringServer := &fakeMonitoringServer{}
+	newServerFn := func() (reload.ServerController, error) {
+		return monitoringServer, nil
+	}
+	monitoringReloader := reload.NewServerReloader(newServerFn, logger, monitoringCfg.DefaultConfig())
+
+	coord := &Coordinator{
+		logger:           logger,
+		agentInfo:        &info.AgentInfo{},
+		stateBroadcaster: broadcaster.New(State{}, 0, 0),
+		managerChans: managerChans{
+			configManagerUpdate: configChan,
+		},
+		runtimeMgr: runtimeManager,
+		vars:       emptyVars(t),
+	}
+	coord.RegisterMonitoringServer(monitoringReloader)
+
+	// Create a policy with one input and one output
+	cfg := config.MustNewConfigFrom(`
+outputs:
+  default:
+    type: elasticsearch
+inputs:
+  - id: test-input
+    type: filestream
+    use_output: default
+`)
+
+	// Send the policy change and make sure it was acknowledged.
+	cfgChange := &configChange{cfg: cfg}
+	configChan <- cfgChange
+	coord.runLoopIteration(ctx)
+	assert.True(t, cfgChange.acked, "Coordinator should ACK a successful policy change")
+
+	// server is started by default
+	assert.True(t, monitoringServer.startTriggered)
+	assert.True(t, monitoringServer.isRunning)
+
+	// disable monitoring
+	cfgDisableMonitoring := config.MustNewConfigFrom(`
+agent.monitoring.enabled: false
+outputs:
+  default:
+    type: elasticsearch
+inputs:
+  - id: test-input
+    type: filestream
+    use_output: default
+`)
+
+	// Send the policy change and make sure it was acknowledged.
+	monitoringServer.Reset()
+	cfgChange = &configChange{cfg: cfgDisableMonitoring}
+	configChan <- cfgChange
+	coord.runLoopIteration(ctx)
+	assert.True(t, cfgChange.acked, "Coordinator should ACK a successful policy change")
+
+	// server is started by default
+	assert.True(t, monitoringServer.stopTriggered)
+	assert.False(t, monitoringServer.isRunning)
+
+	// enable monitoring
+	cfgEnabledMonitoring := config.MustNewConfigFrom(`
+agent.monitoring.enabled: true
+outputs:
+  default:
+    type: elasticsearch
+inputs:
+  - id: test-input
+    type: filestream
+    use_output: default
+`)
+
+	// Send the policy change and make sure it was acknowledged.
+	monitoringServer.Reset()
+	cfgChange = &configChange{cfg: cfgEnabledMonitoring}
+	configChan <- cfgChange
+	coord.runLoopIteration(ctx)
+	assert.True(t, cfgChange.acked, "Coordinator should ACK a successful policy change")
+
+	// server is started by default
```

Comment:
Comment should be changed since we are explicitly enabling monitoring in the policy now (as opposed to it being enabled by default)?